File: uikit/src/main/java/io/agora/chat/uikit/chat/interfaces/IChatEmojiconMenu.java
package io.agora.chat.uikit.chat.interfaces;



import java.util.List;

import io.agora.chat.uikit.models.EaseEmojiconGroupEntity;

public interface IChatEmojiconMenu {
    /**
     * Add emoticon group
     * @param groupEntity
     */
    void addEmojiconGroup(EaseEmojiconGroupEntity groupEntity);

    /**
     * Add emoticons list
     * @param groupEntitieList
     */
    void addEmojiconGroup(List<EaseEmojiconGroupEntity> groupEntitieList);

    /**
     * Remove emoticon group
     * @param position
     */
    void removeEmojiconGroup(int position);

    /**
     * Set whether the TabBar is visible
     * @param isVisible
     */
    void setTabBarVisibility(boolean isVisible);

    /**
     * Set up emoticon monitoring
     * @param listener
     */
    void setEmojiconMenuListener(EaseEmojiconMenuListener listener);
}

File: ease-im-kit/src/main/kotlin/io/agora/uikit/feature/chat/interfaces/IChatEmojiconMenu.kt
package io.agora.uikit.feature.chat.interfaces

import io.agora.uikit.model.EaseEmojiconGroupEntity
interface IChatEmojiconMenu {
    /**
     * Add emoticon group
     * @param groupEntity
     */
    fun addEmojiconGroup(groupEntity: EaseEmojiconGroupEntity)

    /**
     * Add emoticons list
     * @param groupEntitieList
     */
    fun addEmojiconGroup(groupEntitieList: List<EaseEmojiconGroupEntity>?)

    /**
     * Remove emoticon group
     * @param position
     */
    fun removeEmojiconGroup(position: Int)

    /**
     * Set whether the TabBar is visible
     * @param isVisible
     */
    fun setTabBarVisibility(isVisible: Boolean)

    /**
     * Set up emoticon monitoring
     * @param listener
     */
    fun setEmojiconMenuListener(listener: EaseEmojiconMenuListener?)
}